lib/postgres/postgres-pr/message.rb

```diff
@@ -46,7 +46,14 @@ def self.read(stream, startup=false)
     type = stream.read_exactly_n_bytes(1) unless startup
     length = stream.read_exactly_n_bytes(4).to_s.unpack('N').first  # FIXME: length should be signed, not unsigned
 
-    raise ParseError unless length >= 4
+    raise ParseError unless (!length.nil? && length >= 4)
```

Comment:
I am not a super fan of anything involving `unless` and `not`  Unless is bad enough, but we're a happy Ruby family [twitch]
Could this be rewritten as:
`raise ParseError if(length.nil? || length < 4)`?

Reply:
Yep that makes sense, I was trying to keep the original `unless` statement in there to minimize impact but this makes sense.

Reply:
Should now be fixed